src/controller_examples/rabbitmq_controller/proof/resource.rs

```diff
@@ -0,0 +1,100 @@
+        ResourceKind::ServiceAccount => ServiceAccountBuilder::make_view(rabbitmq, state),
+        ResourceKind::Role => RoleBuilder::make_view(rabbitmq, state),
+        ResourceKind::RoleBinding => RoleBindingBuilder::make_view(rabbitmq, state),
+        ResourceKind::StatefulSet => StatefulSetBuilder::make_view(rabbitmq, state),
+    }
+}
+
+pub open spec fn update_view(kind: ResourceKind, rabbitmq: RabbitmqClusterView, state: RabbitmqReconcileStateView, obj: DynamicObjectView) -> Result<DynamicObjectView, RabbitmqError> {
+    match kind {
+        ResourceKind::HeadlessService => HeadlessServiceBuilder::update_view(rabbitmq, state, obj),
+        ResourceKind::Service => ServiceBuilder::update_view(rabbitmq, state, obj),
+        ResourceKind::ErlangCookieSecret => ErlangCookieBuilder::update_view(rabbitmq, state, obj),
+        ResourceKind::DefaultUserSecret => DefaultUserSecretBuilder::update_view(rabbitmq, state, obj),
+        ResourceKind::PluginsConfigMap => PluginsConfigMapBuilder::update_view(rabbitmq, state, obj),
+        ResourceKind::ServerConfigMap => ServerConfigMapBuilder::update_view(rabbitmq, state, obj),
+        ResourceKind::ServiceAccount => ServiceAccountBuilder::update_view(rabbitmq, state, obj),
+        ResourceKind::Role => RoleBuilder::update_view(rabbitmq, state, obj),
+        ResourceKind::RoleBinding => RoleBindingBuilder::update_view(rabbitmq, state, obj),
+        ResourceKind::StatefulSet => StatefulSetBuilder::update_view(rabbitmq, state, obj),
+    }
+}
+
+pub open spec fn state_after_create_or_update_view(kind: ResourceKind, obj: DynamicObjectView, state: RabbitmqReconcileStateView) -> (res: Result<RabbitmqReconcileStateView, RabbitmqError>) {
+    match kind {
+        ResourceKind::HeadlessService => HeadlessServiceBuilder::state_after_create_or_update_view(obj, state),
+        ResourceKind::Service => ServiceBuilder::state_after_create_or_update_view(obj, state),
+        ResourceKind::ErlangCookieSecret => ErlangCookieBuilder::state_after_create_or_update_view(obj, state),
+        ResourceKind::DefaultUserSecret => DefaultUserSecretBuilder::state_after_create_or_update_view(obj, state),
+        ResourceKind::PluginsConfigMap => PluginsConfigMapBuilder::state_after_create_or_update_view(obj, state),
+        ResourceKind::ServerConfigMap => ServerConfigMapBuilder::state_after_create_or_update_view(obj, state),
+        ResourceKind::ServiceAccount => ServiceAccountBuilder::state_after_create_or_update_view(obj, state),
+        ResourceKind::Role => RoleBuilder::state_after_create_or_update_view(obj, state),
+        ResourceKind::RoleBinding => RoleBindingBuilder::state_after_create_or_update_view(obj, state),
+        ResourceKind::StatefulSet => StatefulSetBuilder::state_after_create_or_update_view(obj, state),
+    }
+}
+
+pub open spec fn resource_state_matches(kind: ResourceKind, rabbitmq: RabbitmqClusterView, obj: DynamicObjectView) -> bool {
```

Comment:
The term `ResourceKind` is a bit confusing since we have a `Kind`. Can you rename it to `SubResource`? And also rename the variable `kind` to `sub_resource?